packages/snaps-utils/src/json-rpc.ts

```diff
@@ -103,7 +128,11 @@ export function isOriginAllowed(
   }
 
   // If the origin is in the `allowedOrigins` list, it is allowed.
-  if (origins.allowedOrigins?.includes(origin)) {
+  if (
+    origins.allowedOrigins
+      ?.map(createOriginRegExp)
+      .some((regex) => regex.test(origin))
```

Comment:
To avoid creating regex unnecessarily, what do you think of
```ts
function checkOrigin(origin: string, originSpecifier: string) {
  if (originSpecifier === '*') {
    return true;
  }

  // Create regex and test.
}


// ...
origins.allowedOrigins?.some((specifier) => checkOrigin(origin, specifier))
```

Reply:
Yeah, I added one more optimization to this